sos/report/plugins/container_log.py

```diff
@@ -9,26 +9,51 @@
 # See the LICENSE file in the source distribution for further information.
 
 import os
-from sos.report.plugins import Plugin, IndependentPlugin
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
 
 
 class ContainerLog(Plugin, IndependentPlugin):
 
     short_desc = 'All logs under /var/log/containers'
     plugin_name = 'container_log'
     logdir = '/var/log/containers/'
+    poddir = '/var/log/pods/'
     files = (logdir, )
 
+    option_list = [
+        PluginOpt('rotated', default=False, val_type=bool,
+                  desc='get all logs from /var/log/pods regardless of size, '
+                  'which will include all rotated logs'),
+        PluginOpt('maxage', default=None, val_type=int,
+                  desc='gather only logs with `mtime` not older than this many'
+                  ' hours')
+    ]
+
     def setup(self):
+        maxage = self.get_option('maxage')
+        sizelimit = None
+        # Remove size limit from containers logs when getting rotated, since
+        # they are just symlinks to the pods logs
```

Comment:
Our sizelimiting takes the link's destination into account when checking sizes for collections, so I'm not sure what the implication is here? We definitely do not want to have a wide open "collect all logs regardless of sizes" default posture here. We've had that bite us in the rear in the past several times - no one likes several GB sos reports.

Reply:
I tried, and failed, to explain it in the PR description.

The idea is that if you set the `rotated` option but you don't change the limits at all you are going to either get the exact same content as without the option (because like you said getting the symlinks checks the actual destination files) or we are going to get logs from the pods that don't have a symlink (not necessarily rotated logs), in both cases it seems wrong.

Another option is for the `rotated` option to ignore all the `.log` files from the pods directory and only get rotated logs instead.

This later options sounds reasonable to me and in line with the option name.  I'll update the PR.